### ISSUE TITLE
how to use yolo11 inference output as input for another yolo11 model

### ISSUE THREAD

**Issue by sarabhan** [ultralytics/ultralytics#22809]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello! 

I am working on a helmet & safety vest detection for construction workers. I want to try identifying people first using a yolo model, and then where ever a person is identified in the frame, i want to run another model that detects if the person has a helmet and vest.

However, I need help in understanding how do i exactly pass the first model's inference output into the second model as input. 
this is how i run the model on my cctv footage video

```
# Load YOLO model
model = YOLO(MODEL_PATH)
model.to(DEVICE)

# Video capture
cap = cv2.VideoCapture(VIDEO_SOURCE)
if not cap.isOpened():
    raise RuntimeError(f"Cannot open video source {VIDEO_SOURCE}")

fps = cap.get(cv2.CAP_PROP_FPS) or 30
w = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
h = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))

writer = None
if OUT_VIDEO:
    fourcc = cv2.VideoWriter_fourcc(*"mp4v")
    writer = cv2.VideoWriter(OUT_VIDEO, fourcc, fps, (w, h))

print("Starting YOLO video inference. Press 'q' to quit.")

while True:
    ret, frame = cap.read()
    if not ret:
        break

    # YOLO inference (confidence & IoU thresholds)
    results = model.predict(
        frame, 
        conf=CONF_THRESH,
        iou=IOU_THRESH,
        verbose=False
    )

    # Draw predictions on the frame
    annotated_frame = results[0].plot()

    # Display smaller window
    display_scale = 0.30
    display_frame = cv2.resize(annotated_frame, None, fx=display_scale, fy=display_scale)

    cv2.imshow("YOLO11 Inference", display_frame)

    if writer:
        writer.write(annotated_frame)

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
if writer:
    writer.release()
cv2.destroyAllWindows()
```

should i be passing just the bounding box co-ordinates? or should i pass the cropped image of the person? if yes, then how should i proceed with this?

if no, what is the alternative? and how should i proceed with that? if any additional information is required to get a better answer, let me know!

please help!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sarabhan, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly — an Ultralytics engineer will also review your issue and assist you soon.

Your question about chaining YOLO11 models (person detection first, then PPE like helmet/vest) is a common multi-stage pipeline pattern, and more implementation details (your current code, environment, and how you’re running inference) will help make any guidance more specific. If you can, please share:
- How you are currently running YOLO11 (Python script, notebook, CLI, Ultralytics HUB, etc.)
- Relevant code snippets showing how you run the first model’s inference
- A brief description or example of your desired end-to-end workflow (per-frame, per-image, real-time, batch, etc.)

We also recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered ⭐

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training or pipeline design ❓ Question (like your multi-model PPE detection use case), please provide as much information as possible, including:
- Example images or a description of your dataset
- How your models are trained (classes, input size, confidence thresholds, etc.)
- Any existing code for how you’re trying to connect model 1 (person) to model 2 (helmet/vest)
And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Before proceeding, please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to verify any issue you might encounter is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You will have to crop the image with boxes and then make inference on the cropped image

**Comment by sarabhan** [comment]
im afraid i didnt understand your approach clearly, do you mind explaining in detail?

**Comment by glenn-jocher** [comment]
Nice use case. The second model always needs an image (or batch of images), so you should crop the person regions from the first model’s detections and pass those crops, not just the coordinates.

Conceptually you run a “person detector” on the full frame, then for each box you slice the NumPy frame and call the PPE model on that slice:

```python
from ultralytics import YOLO
import cv2

person_model = YOLO("person_detector.pt")   # detects only 'person'
ppe_model    = YOLO("ppe_detector.pt")      # detects helmet/vest

while True:
    ret, frame = cap.read()
    if not ret:
        break

    # 1) detect persons
    person_res = person_model.predict(
        frame,
        conf=CONF_THRESH,
        iou=IOU_THRESH,
        verbose=False,
    )[0]

    # 2) for each person box, crop and run PPE model
    for box in person_res.boxes:
        x1, y1, x2, y2 = box.xyxy[0].int().tolist()
        person_crop = frame[y1:y2, x1:x2]

        ppe_res = ppe_model.predict(
            person_crop,
            conf=CONF_PPE,
            verbose=False,
        )[0]

        # here you can draw PPE boxes back on the original frame
        # by adding (x1, y1) as an offset to each ppe_res.boxes.xyxy

    # ...display / save frame as you already do
```

If your first model is COCO-pretrained instead of “person-only,” just filter `person_res.boxes` by the class index for `person` before cropping. For more patterns (including a higher‑level helper class) have a look at the [Object cropping guide](https://docs.ultralytics.com/guides/object-cropping/).

**Comment by sarabhan** [comment]
thank you!

**Comment by glenn-jocher** [comment]
Glad it helped — if you run into any issues wiring the two models together (e.g., mapping PPE boxes back to the full frame), feel free to share a small code snippet here and we can help refine it.

**Comment by lo-bei-ve** [comment]
I’m sure an LLM can give you excellent help with this—whether it’s ChatGPT, Gemini, or DeepSeek. Learn to use AI and boost your development efficiency.

**Comment by glenn-jocher** [comment]
AI tools can definitely speed things up, but for Ultralytics‑specific questions (like chaining two YOLO11 models) it’s still worth confirming details here, since we can align answers with the current APIs and behavior—if you or @sarabhan hit any concrete issues with the two‑stage setup, feel free to share a minimal snippet and we can help debug.
